protocol/messenger_profile_showcase.go

```diff
@@ -71,69 +86,102 @@ func toProfileShowcaseCollectibleProto(preferences []*ProfileShowcaseCollectible
+	entries := []*protobuf.ProfileShowcaseVerifiedToken{}
 	for _, preference := range preferences {
 		if preference.ShowcaseVisibility != visibility {
 			continue
 		}
 
-		tokens = append(tokens, &protobuf.ProfileShowcaseVerifiedToken{
+		entries = append(entries, &protobuf.ProfileShowcaseVerifiedToken{
 			Symbol: preference.Symbol,
 			Order:  uint32(preference.Order),
 		})
 	}
-	return tokens
+	return entries
 }
 
-func toProfileShowcaseUnverifiedTokensProto(preferences []*ProfileShowcaseUnverifiedTokenPreference, visibility ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseUnverifiedToken {
-	tokens := []*protobuf.ProfileShowcaseUnverifiedToken{}
+func (m *Messenger) toProfileShowcaseUnverifiedTokensProto(preferences []*identity.ProfileShowcaseUnverifiedTokenPreference, visibility identity.ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseUnverifiedToken {
+	entries := []*protobuf.ProfileShowcaseUnverifiedToken{}
 	for _, preference := range preferences {
 		if preference.ShowcaseVisibility != visibility {
 			continue
 		}
 
-		tokens = append(tokens, &protobuf.ProfileShowcaseUnverifiedToken{
+		entries = append(entries, &protobuf.ProfileShowcaseUnverifiedToken{
 			ContractAddress: preference.ContractAddress,
 			ChainId:         preference.ChainID,
 			Order:           uint32(preference.Order),
 		})
 	}
-	return tokens
+	return entries
 }
 
-func fromProfileShowcaseCommunityProto(messages []*protobuf.ProfileShowcaseCommunity) []*ProfileShowcaseCommunity {
-	communities := []*ProfileShowcaseCommunity{}
-	for _, entry := range messages {
-		communities = append(communities, &ProfileShowcaseCommunity{
-			CommunityID: entry.CommunityId,
-			Order:       int(entry.Order),
+func (m *Messenger) fromProfileShowcaseCommunityProto(senderPubKey *ecdsa.PublicKey, messages []*protobuf.ProfileShowcaseCommunity) []*identity.ProfileShowcaseCommunity {
+	entries := []*identity.ProfileShowcaseCommunity{}
+	for _, message := range messages {
+		entry := &identity.ProfileShowcaseCommunity{
+			CommunityID: message.CommunityId,
+			Order:       int(message.Order),
+		}
+
+		community, err := m.FetchCommunity(&FetchCommunityRequest{
+			CommunityKey:    message.CommunityId,
+			Shard:           nil,
+			TryDatabase:     true,
+			WaitForResponse: true,
 		})
+		if err != nil {
+			m.logger.Warn("failed to fetch community for profile entry ", zap.Error(err))
+		}
+
+		if community != nil && community.Encrypted() {
+			grant, err := community.VerifyGrantSignature(message.Grant)
+			if err != nil {
+				m.logger.Warn("failed to verify grant signature ", zap.Error(err))
+				entry.MembershipStatus = identity.ProfileShowcaseMembershipStatusNotAMember
+			} else {
+				if grant != nil && bytes.Equal(grant.MemberId, crypto.CompressPubkey(senderPubKey)) {
```

Comment:
@osmaczko i'm not sure about `grant.MemberId`, is this member id or contol node's id

Reply:
It is member id.